llvm/tools/sycl-post-link/DeviceLibFunctions.h

```diff
@@ -0,0 +1,62 @@
+//===- DeviceLibFunctions.h - record the functions in each device library--===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifndef __DEVICELIB_FUNCS_LIST__
+#define __DEVICELIB_FUNCS_LIST__
+#include <string>
+// All __devicelib_* functions must be sorted in alphabetic order as we
```

Comment:
This looks like a source of obscure bugs. Suggest that this sorting requirement is dropped and the tool sorts automatically at startup. Or creates a map and uses map lookup instead of binary_search.

Reply:
Hi, @kbobrovs 
I have updated the patch to use map instead of sorted array.